src/Orleans/Core/GrainExtensions.cs

```diff
@@ -62,6 +62,16 @@ public static TGrainInterface Cast<TGrainInterface>(this IAddressable grain)
             return RuntimeClient.Current.InternalGrainFactory.Cast<TGrainInterface>(grain);
         }
 
+        /// <summary>
+        /// Binds the grain reference to the provided <see cref="IGrainFactory"/>.
+        /// </summary>
+        /// <param name="grain">The grain reference.</param>
+        /// <param name="grainFactory">The grain factory.</param>
+        public static void BindGrainReference(this IAddressable grain, IGrainFactory grainFactory)
```

Comment:
This is a convenience method to allow grains to be bound to the grain factory. As you can see, it's trivial. Should it be removed?